module/mempool/stdmap/incorporated_result_seals.go

```diff
@@ -1,26 +1,109 @@
 package stdmap
 
 import (
+	"fmt"
+
 	"github.com/onflow/flow-go/model/flow"
 	"github.com/onflow/flow-go/module/mempool"
 )
 
+type sealSet map[flow.Identifier]*flow.IncorporatedResultSeal
+
 // IncorporatedResultSeals implements the incorporated result seals memory pool
 // of the consensus nodes, used to store seals that need to be added to blocks.
 type IncorporatedResultSeals struct {
 	*Backend
+	// index the seals by the height of the executed block
+	byHeight     map[uint64]sealSet
+	lowestHeight uint64
+}
+
+func indexByHeight(seal *flow.IncorporatedResultSeal) uint64 {
+	return seal.Header.Height
+}
+
+// NewIncorporatedResultSeals creates a mempool for the incorporated result seals
+func NewIncorporatedResultSeals(limit uint) *IncorporatedResultSeals {
+	byHeight := make(map[uint64]sealSet)
+
+	// assuming all the entities are for unsealed blocks, then we will remove a seal
+	// with the largest height.
+	ejector := func(entities map[flow.Identifier]flow.Entity) (flow.Identifier, flow.Entity) {
+		maxHeight := uint64(0)
+		var sealsAtMaxHeight sealSet
+		for height, seals := range byHeight {
+			if height > maxHeight {
+				maxHeight = height
+				sealsAtMaxHeight = seals
+			}
+		}
+		if len(sealsAtMaxHeight) == 0 {
+			// this can only happen if mempool is empty or if the secondary index was inconsistently updated
+			panic("cannot eject element from empty mempool")
+		}
```

Comment:
I don't think we need both this conditional and the `panic` on line 48, they will have the same effect individually